terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/runners.ts

```diff
@@ -514,142 +514,58 @@ export async function tryReuseRunner(
-          // is present and it's less than 5 mins.
-          //
-          // Stage: record the stage of the runner, in this case, it's in the
-          // RunnerReplaceEBSVolume.  Refresh and scaleup pipelines will not
-          // reuse the runner if the Stage is present and it's RunnerReplaceEBSVolume.
-          // the stage tag will be removed once the replace volume task is completed
-          // at job's startup.sh
-          await expBackOff(() => {
-            return metrics.trackRequestRegion(
-              runner.awsRegion,
-              metrics.ec2CreateTagsAWSCallSuccess,
-              metrics.ec2CreateTagsAWSCallFailure,
-              () => {
-                return ec2
-                  .createTags({
-                    Resources: [runner.instanceId],
-                    Tags: [
-                      { Key: 'EBSVolumeReplacementRequestTm', Value: `${Math.floor(Date.now() / 1000)}` },
-                      { Key: 'Stage', Value: 'RunnerReplaceEBSVolume' },
-                    ],
-                  })
-                  .promise();
-              },
-            );
-          });
+          await createTagForReuse(ec2, runner, metrics);
           console.debug(`[tryReuseRunner]: Reuse of runner ${runner.instanceId}: Created reuse tag`);
 
           // Delete EphemeralRunnerFinished tag to make sure other pipelines do not
           // pick this instance up since it's in next stage, in this case, it's in the ReplaceVolume stage.
-          await expBackOff(() => {
-            return metrics.trackRequestRegion(
-              runner.awsRegion,
-              metrics.ec2DeleteTagsAWSCallSuccess,
-              metrics.ec2DeleteTagsAWSCallFailure,
-              () => {
-                return ec2
-                  .deleteTags({
-                    Resources: [runner.instanceId],
-                    Tags: [{ Key: 'GithubRunnerID' }, { Key: 'EphemeralRunnerFinished' }],
-                  })
-                  .promise();
-              },
-            );
-          });
+          await deleteTagForReuse(ec2, runner, metrics);
           console.debug(`[tryReuseRunner]: Reuse of runner ${runner.instanceId}: Tags deleted`);
 
-          await expBackOff(() => {
-            return metrics.trackRequestRegion(
-              runner.awsRegion,
-              metrics.ec2CreateReplaceRootVolumeTaskSuccess,
-              metrics.ec2CreateReplaceRootVolumeTaskFailure,
-              () => {
-                return ec2
-                  .createReplaceRootVolumeTask({
-                    InstanceId: runner.instanceId,
-                    DeleteReplacedRootVolume: true,
-                  })
-                  .promise();
-              },
-            );
-          });
+          await replaceRootVolume(ec2, runner, metrics);
           console.debug(`[tryReuseRunner]: Reuse of runner ${runner.instanceId}: Replace volume task created`);
 
           await addSSMParameterRunnerConfig(
```

Comment:
as you are in this task, why not move this guy to another function as well, just to help improve the looks and reads of the spaguetti code I wrote? :)